tests/integration.rs

```diff
@@ -2264,6 +2279,48 @@ mod tests {
             });
         }
 
+        #[cfg(target_arch = "x86_64")]
+        #[cfg_attr(not(feature = "mmio"), test)]
+        fn test_guest_numa_nodes() {
+            test_block!(tb, "", {
+                let mut focal = UbuntuDiskConfig::new(FOCAL_IMAGE_NAME.to_string());
+                let guest = Guest::new(&mut focal);
+                let mut cmd = GuestCommand::new(&guest);
+                cmd.args(&["--cpus", "boot=1"])
+                    .args(&["--memory", "size=0"])
+                    .args(&[
+                        "--memory-zone",
+                        "size=1G,guest_numa_node=0",
+                        "size=2G,guest_numa_node=1",
+                        "size=3G,guest_numa_node=2",
```

Comment:
Question: I assume the `guest_numa_node` (e.g. `Proximity Domain` in SRAT) from user has to be starting from `0` and be continuous, e.g. the following inputs should be invalid: `--memory-zone size=1G,guest_numa_node=1 size=3G,guest_numa_node=3`. If that's true, we may need to add a sanity check when we initialize the `numa_nodes` in `MemoryManager`?

Reply:
Well it's not entirely invalid. From an ACPI perspective, you can have proximity domains not starting at 0 and not contiguous, the VM would still boot, and you would see some logs in the dmesg saying that proximity domain 5 has been assigned to node 0, and proximity domain 7 to node 1.
So I'd say it's highly discouraged to use such weird proximity domains numbers, but it wouldn't break anything doing so.